test/models/detector/ml_citation_test.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Detector
+  class MlCitationTest < ActiveSupport::TestCase
+    # Ensure the expected_env? method returns false without everything set.
+    test 'expected_env? returns true when all three env values are defined' do
+      with_enabled_mlcitation do
+        path = ENV.fetch('DETECTOR_LAMBDA_PATH', nil)
+        secret = ENV.fetch('DETECTOR_LAMBDA_CHALLENGE_SECRET', nil)
+        url = ENV.fetch('DETECTOR_LAMBDA_URL', nil)
+
+        assert(path && secret && url)
+        assert_predicate(Detector::MlCitation, :expected_env?)
+      end
+    end
+
+    test 'expected_env? returns false without DETECTOR_LAMBDA_CHALLENGE_SECRET set' do
+      with_enabled_mlcitation do
+        assert_predicate(Detector::MlCitation, :expected_env?)
+        ClimateControl.modify DETECTOR_LAMBDA_CHALLENGE_SECRET: nil do
+          assert_equal false, Detector::MlCitation.expected_env?
+        end
+      end
+    end
+
+    test 'expected_env? returns false without DETECTOR_LAMBDA_PATH set' do
+      with_enabled_mlcitation do
+        assert_predicate(Detector::MlCitation, :expected_env?)
+        ClimateControl.modify DETECTOR_LAMBDA_PATH: nil do
+          assert_equal false, Detector::MlCitation.expected_env?
+        end
+      end
+    end
+
+    test 'expected_env? returns false without DETECTOR_LAMBDA_URL set' do
+      with_enabled_mlcitation do
+        assert_predicate(Detector::MlCitation, :expected_env?)
+        ClimateControl.modify DETECTOR_LAMBDA_URL: nil do
+          assert_equal false, Detector::MlCitation.expected_env?
+        end
+      end
+    end
+
+    # These next three tests may not be needed, since the expected_env? method is tested above.
```

Comment:
This comment feels a bit fragile as someone could insert a test in between any of these at some point. Maybe either drop the tests you feel are redundant or add the a more focused comment to each test such as:
`This test may not be needed, since the expected_env? method is also tested explicitly`